src/Generator/ValueGenerator.php

```diff
@@ -312,12 +316,19 @@ public function generate()
             $type = $this->getAutoDeterminedType($value);
         }
 
-        if ($type == self::TYPE_ARRAY) {
+        if (in_array($type, array(self::TYPE_ARRAY, self::TYPE_ARRAY_LONG, self::TYPE_ARRAY_SHORT))) {
             foreach ($value as &$curValue) {
                 if ($curValue instanceof self) {
                     continue;
                 }
-                $curValue = new self($curValue, self::TYPE_AUTO, self::OUTPUT_MULTIPLE_LINE, $this->getConstants());
+
+                if (is_array($curValue) && $type == self::TYPE_ARRAY_SHORT) {
+                    $newType = self::TYPE_ARRAY_SHORT;
```

Comment:
I see a bit optimization here.

Previously we have calculated if `$type` is any of the allowed array types. 

So:

```php

$isArrayType = in_array($type, [self::TYPE_ARRAY, self::TYPE_ARRAY_LONG, self::TYPE_ARRAY_SHORT]);

if ($isArrayType) {
  //...
  if (is_array($curValue)) {
    $newType = $type;
  } else {
    $newType = self::TYPE_AUTO;
  }
  //...
}